lib/in_stock/application.ex

```diff
@@ -26,6 +25,33 @@ defmodule InStock.Application do
     Supervisor.start_link(children, opts)
   end
 
+  # Returns a child spec for an ad-hoc supervisor that will start the dynamic
+  # supervisor CheckerSupervisor and then start its initial children
+  #
+  # References:
+  # - https://elixirforum.com/t/understanding-dynamicsupervisor-no-initial-children/14938/2
+  # - https://github.com/slashdotdash/til/blob/master/elixir/dynamic-supervisor-start-children.md
+  defp checker_child_spec() do
+    checker_children = [
+      InStock.Checker.CheckerSupervisor,
+      {Task, &InStock.Checker.CheckerSupervisor.spawn_checkers/0}
+    ]
+
+    checker_opts = [
+      strategy: :rest_for_one,
+      name: InStock.Checker.CheckerStartSupervisor
+    ]
```

Comment:
rest for one strategy makes sure that the task cannot run or live unless the child before it (the dynamic sup) is up and running